Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java

```diff
@@ -195,6 +213,54 @@ private void deleteMetadataDirectory(Process process) throws IOException {
         fileService.deleteProcessContent(process);
     }
 
+
+    private void generateImages(List<Process> processes, GenerationMode generationMode, List<String> folders) {
+        for (Process process : processes) {
+            Folder generatorSource = process.getProject().getGeneratorSource();
+            if (Objects.isNull(generatorSource)) {
+                Helper.setErrorMessage("kitodoScript.generateImages.error.noSourceFolder",
+                    new String[] {process.getTitle(), process.getProject().getTitle() });
+                continue;
+            }
+            Subfolder sourceFolder = new Subfolder(process, generatorSource);
+            if (sourceFolder.listContents().isEmpty()) {
+                Helper.setErrorMessage("kitodoScript.generateImages.error.noSourceFiles",
+                    new String[] {process.getTitle(), sourceFolder.getRelativeDirectoryPath() });
+                continue;
+            }
+            List<String> ungeneratableFolders = new ArrayList<>(folders);
+            boolean all = folders.size() == 1 && folders.get(0).equalsIgnoreCase("all");
+            List<Subfolder> outputs = new ArrayList<>();
+            for (Folder folder : process.getProject().getFolders()) {
+                if ((all || folders.contains(folder.getPath())) && !folder.equals(generatorSource)
+                        && (folder.getDerivative().isPresent() || folder.getDpi().isPresent()
+                                || folder.getImageScale().isPresent() || folder.getImageSize().isPresent())) {
+                    outputs.add(new Subfolder(process, folder));
+                    ungeneratableFolders.remove(folder.getPath());
+                }
+            }
+            if (outputs.isEmpty()) {
+                Helper.setErrorMessage("kitodoScript.generateImages.error.noDestination",
+                    new String[] {process.getTitle(), String.join(", ", ungeneratableFolders) });
+                continue;
+            }
+            ImageGenerator imageGenerator = new ImageGenerator(sourceFolder, generationMode, outputs);
+            TaskManager.addTask(new TaskImageGeneratorThread(process.getTitle(), imageGenerator));
+            String generationModeTranslated = Helper
+                    .getTranslation("imageGenerator.generationMode.".concat(generationMode.toString()));
+            String generatableFolders = outputs.stream().map(Subfolder::getFolder).map(Folder::getPath)
```

Comment:
actually the folders are not generated, right? They are filled with images?

Reply:
Folders are never _generated_ (they can be _created_, but that's not the topic here), only the contents in the folders are ever generated.